homeassistant/components/rainmachine/util.py

```diff
@@ -35,6 +37,39 @@ class RunStates(StrEnum):
 }
 
 
+@callback
+def async_clean_up_old_entities(
+    hass: HomeAssistant, entry: ConfigEntry, unique_id_suffixes_to_remove: Iterable[str]
+) -> None:
+    """Clean up old, no-longer-used entities."""
+    ent_reg = entity_registry.async_get(hass)
+    for entity_registry_item in [
+        e
+        for e in ent_reg.entities.values()
+        if e.config_entry_id == entry.entry_id
+        and any(
+            suffix
+            for suffix in unique_id_suffixes_to_remove
+            if e.unique_id.endswith(suffix)
+        )
```

Comment:
`endswith` takes a tuple to match on. No need to implement this logic yourself.

Reply:
I never knew this! 😱